src/main/java/com/scaleunlimited/flinkcrawler/topology/CrawlTopologyBuilder.java

```diff
@@ -399,6 +422,26 @@ public FetchUrl map(Tuple3<CrawlStateUrl, FetchUrl, FetchUrl> justHasFetchUrl)
                                 TimeUnit.SECONDS, 10000)
                         .name("FetchUrlsFunction");
 
+        // Save off the content by converting it to a WARC record and passing it on to the provided
+        // content sink function.
+        DataStream<FetchResultUrl> fetchResultUrlsToSave = fetchResultUrls;
+        SingleOutputStreamOperator<Tuple2<NullWritable, WARCWritable>> warcStream = fetchResultUrlsToSave
+                .flatMap(new CreateWARCWritableFunction(_userAgentString))
+                .name("Create WARC writable");
+        
+        DataStreamSink<Tuple2<NullWritable, WARCWritable>> contentSinkUsingOutputFormat = null;
+        if (_contentSink != null) {
+            contentSinkUsingOutputFormat = warcStream.addSink(_contentSink);
+        } else {
+            Job job = Job.getInstance();
+            WARCOutputFormat.setOutputPath(job, new Path(_contentFilePathString));
```

Comment:
You're going to have to explain this line of code to me. I'm mystified. It looks like a static method call, but it's not a static method of WARCOutputFormat itself.

Reply:
This is just the way it is done in Hadoop world. 

     FileOutputFormat.setOutputPath()
Since WARCOutputFormat extends FileOutputFormat, I am using WARCOutputFormat.setOutputPath.

Reply:
OK